packages/components/src/segmented-control/component.tsx

```diff
@@ -128,4 +126,32 @@ function getShowSeparator( radio: any, index: number ) {
 	return showSeparator;
 }
 
-export default contextConnect( SegmentControl, 'SegmentControl' );
+function ControlOption(
+	props: PolymorphicComponentProps< SegmentedControlOption, 'input' >,
+	forwardedRef: import('react').Ref< any >
+) {
+	const radio = useContext( RadioContext );
+	const buttonProps = useContextSystem( props, 'SegmentedControlOption' );
+	const index = radio.items.findIndex(
+		( item: any ) => item.id === buttonProps.id
+	);
+	const showSeparator = getShowSeparator( radio, index );
+	return (
+		<Button
+			ref={ forwardedRef }
+			{ ...{ ...radio, ...buttonProps, showSeparator } }
+		/>
+	);
+}
```

Comment:
Wow! 🤩

Reply:
I think we should split the ControlOption into a separate module instead of putting it all in one.

Reply:
I'll still export this as `SegmentedControl.Option` though, no?

Reply:
Yup!